resources/config/varnish/fos_purge_xkey.vcl

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import xkey;
+
+sub fos_purge_recv {
+    if (req.method == "PURGE") {
+        if (!client.ip ~ invalidators) {
+            return (synth(405, "Not allowed"));
+        }
+
+        if (req.http.key) {
```

Comment:
should we call the header `X-Cache-Keys` instead of `key`? that would distinguish between the key on the response and the information which keys to invalidate, and we would avoid to require changing the tag header configuration depending on the tag invalidation mode.

Reply:
you mean `X-Cache-Tags`, yes we can. On responses the header is `xkey`, that's afaik hardcoded in the module, but on purge we have full controll.

Reply:
That said, it's format is different _(space separated and not comma)_ and xkey doc suggest using `xkey-purge` header. Could use `xkey-purge` and `xkey-softpurge` headers to separate hard and soft purging for instance?

Reply:
good point, lets use xkey-purge then, as the doc suggests. and xkey-softpurge for soft purge, so that the user can include one snippet and the php application can chose which mode to do.